### ISSUE TITLE
`bundle update` doesn't show post-install messages

### ISSUE THREAD

**Issue by frostblooded** [rubygems/rubygems#3274]
When I run `bundle install`, my gem's post-install message is shown as it should be, but it doesn't appear when I run `bundle update`. Is this intentional? I fully expected `bundle update` to also show post install messages because it installs new versions of gems.

**Comment by deivid-rodriguez** [comment]
I just tried this and `bundle update` does print post-install messages after installing new gems.

If the gems are already installed, and `bundle update` is only changing the lockfile to use those, then they are not printed, just like a second `bundle install` wouldn't print post-install messages again.

It all looks expected to me. Let me know if I'm missing something.

**Comment by leastbad** [comment]
I published a stub package `holabad` for testing post install message hinkiness.

I'm running rubygems 3.4.2 and bundler 2.4.2. Please assume that I am running `gem uninstall holabad` and `rm Gemfile.lock` after each step outlined below.

When I `gem install holabad` I see "Heck yeah." as intended.

When I run `bundle add holabad` inside of a Rails app, it installs but there is no post install message.

If I add it to `Gemfile` manually and run `bundle`, there is no message. However, if I run `bundle install`, I see the message.

As a control (for science) if I perform these same steps for `httparty`, I end up partying hard at the same moments.

In summary:
`gem install holabad` or `bundle install` with manual `Gemfile`: **message displayed**
`bundle add holabad` or just `bundle` with manual `Gemfile`: **no message**

IMO, this isn't an ideal DX matrix.

**Comment by deivid-rodriguez** [comment]
No one said the behavior you describe is expected behavior, it sounds like an issue in `bundle add` to me. If you want this fixed/looked at, please report a separate issue with reproducible steps. Thanks.
